src/storage/src/table/state_table.rs

```diff
@@ -102,28 +112,112 @@ impl<S: StateStore> StateTable<S> {
         Ok(())
     }
 
-    pub async fn iter(&self, _pk: Row) -> StorageResult<StateTableRowIter<S>> {
-        todo!()
+    pub async fn iter(&self, epoch: u64) -> StorageResult<StateTableRowIter<'_, S>> {
+        let mem_table_iter = self.mem_table.buffer.iter().peekable();
+        StateTableRowIter::new(
+            self.keyspace.clone(),
+            self.column_descs.clone(),
+            mem_table_iter,
+            self.order_types.clone(),
+            epoch,
+            Arc::new(StateStoreMetrics::unused()),
+        )
+        .await
     }
 }
 
-pub struct StateTableRowIter<S: StateStore> {
+pub struct StateTableRowIter<'a, S: StateStore> {
+    mem_table_iter: Peekable<MemTableIter<'a>>,
     cell_based_iter: CellBasedTableRowIter<S>,
-    mem_table: MemTable,
+    cell_basde_item: Option<(Vec<u8>, Row)>,
```

Comment:
should not use `CellBasedTableRowIter` here for two reasons:

* StateTableRowIter should be short-lived, which won't pin a version for too long time. But `CellBasedTableRowIter` will switch version internally, which is not useful.
* `CellBasedTableRowIter` will wait for epoch by default, but `StateTableRowIter` will be used in streaming executors, which cannot wait for epoch.

Reply:
If so,  how can `StateTableRowIter` get the next row in `CellBasedTable`?

Reply:
Directly scan from keyspace.

Reply:
for 1, should we add `long_term_iter` and `short_term_iter` on `CellBasedTable`? in the Batch query there will also be some short-lived scan requirements such as indexed scan. 
for 2, we can change the `epoch` as an option when new an iterator.